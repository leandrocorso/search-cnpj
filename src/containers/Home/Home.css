div.home {
    display: flex;
    align-items: flex-end;
    height: 100%;
    text-align: center;
    color: #eee;
    background-image: linear-gradient(#967cf0, #a36cf3);
}

div.home div.heading {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    width: 100%;
    height: 50%;
}

div.home .chart-icon {
    width: 1rem;
    height: 1rem;
    margin: 0 auto;
    fill: #eee;
}

div.home div.title {
    font-size: 150%;
}

div.home div.title small {
    font-size: 60%;
}

div.home .btn.default {
    color: #967cf0;
}

div.home .btn.default svg {
    fill: #967cf0;
}