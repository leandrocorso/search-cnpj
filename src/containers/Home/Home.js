import React from 'react';
import styled from 'styled-components';

import { Container, Button, Icon } from '../../components';

const Wrapper = styled.div`
    display: flex;
    align-items: flex-end;
    height: 100%;
    text-align: center;
    color: ${p => p.theme.themeDefault.colors.c6}
    background-image: linear-gradient(${p => p.theme.themeDefault.colors.c1}, ${p => p.theme.themeDefault.colors.c2});
`;

const Heading = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    width: 100%;
    height: 50%;
`;

const ChartIcon = styled(Icon)`
    width: 1rem;
    height: 1rem;
    margin: 0 auto;
    fill: ${p => p.theme.themeDefault.colors.c6};
`;

const Title = styled.p`
    font-size: 150%;

    small {
        font-size: 60%;
    }
`;

const Home = () => (
    <Wrapper>
        <Heading>
            <ChartIcon className="testHeadingIcon" src="line-chart" />
            <Title>
                Cotação de seguros<br />
                <small>Solução inovadora da líder de mercado</small>
            </Title>
            <Container>
                <Button className="testSearch" to="/busca-cnpj" icon="arrow-right2" color="primary">Iniciar</Button>
            </Container>
        </Heading>
    </Wrapper>
);

export default Home;
