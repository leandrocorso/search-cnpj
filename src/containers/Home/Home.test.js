import React from 'react';
import { shallow } from 'enzyme';

import Home from './Home';

describe('Testing Home component', () => {
    
    const wrapper = shallow(<Home />);
    
    test('Home exists', () => {
        expect(wrapper);
    });

    test('Home have a button to search page', () => {
        expect(wrapper.find('.testSearch')).toHaveProp({ to : '/busca-cnpj'});
    });

    test('Should have arrow to right icon into the button to search page', () => {
        expect(wrapper.find('.testSearch')).toHaveProp({ icon : 'arrow-right2'});
    });

    test('Should be primary color for the button to search page', () => {
        expect(wrapper.find('.testSearch')).toHaveProp({ color : 'primary'});
    });

    test('Should have "chart icon" in the heading', () => {
        expect(wrapper.find('.testHeadingIcon')).toHaveProp({ src : 'line-chart'});
    });

    test('Should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

})