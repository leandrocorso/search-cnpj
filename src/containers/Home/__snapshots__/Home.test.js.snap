// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home component Should render correctly 1`] = `
<styled.div>
  <styled.div>
    <Styled(Icon)
      className="testHeadingIcon"
      src="line-chart"
    />
    <styled.p>
      Cotação de seguros
      <br />
      <small>
        Solução inovadora da líder de mercado
      </small>
    </styled.p>
    <styled.div>
      <Button
        className="testSearch"
        color="primary"
        icon="arrow-right2"
        to="/busca-cnpj"
      >
        Iniciar
      </Button>
    </styled.div>
  </styled.div>
</styled.div>
`;
